doc/guides/weight_normalization.rst

```diff
@@ -0,0 +1,19 @@
+Weight normalization
+====================
+
+Let's say that you have a neuron with incoming synaptic weights :math:`w_1, w_2, \ldots, w_n`. A plasticity rule might require that the vector norm :math:`|w|` remains constant.
```

Comment:
It would be good to define more precisely what is to be kept constant. From this definition, I'd assume that the Euclidean norm was meant, but the example below sets the sum of the weights to 1.
Could you add a reference of a concrete plasticity rule that makes use of weight normalization?

Reply:
This model [1] (known as the SORN model) makes use of weight normalization, see equation 6:
1. Lazar, A. et al. (2009). SORN: a Self-organizing Recurrent Neural Network. Frontiers in Computational Neuroscience, 3. https://doi.org/10.3389/neuro.10.023.2009 
2. **spiking implementation of 1**: Klos, C. et al. Bridging structure and function: A model of sequence learning and prediction in primary visual cortex. PLoS Computational Biology, 14(6). https://doi.org/10.1371/journal.pcbi.1006187

I agree that we should be more precise in defining the weight normalization procedure, also maybe it's better to use a variable for the total sum, say **w_total** and not define it equal to **1** as it's done in the little script below. See paper [2], equation 9.